src/components/NewCommentForm.tsx

```diff
@@ -1,8 +1,99 @@
-import React from 'react';
+import React, { useCallback, useState } from 'react';
+import classNames from 'classnames';
+import { CommentData } from '../types/Comment';
+
+type Props = {
+  postId: number;
+  addComment: (comment: CommentData) => void;
+  isCommentAdditionError: boolean;
+  isCommentAdditionLoading: boolean;
+};
+
+export const NewCommentForm: React.FC<Props> = React.memo(({
+  postId,
+  addComment,
+  isCommentAdditionError,
+  isCommentAdditionLoading,
+}) => {
+  const [userName, setUserName] = useState('');
```

Comment:
Consider using a more descriptive name for the state variables, like 'inputName', 'inputEmail', and 'inputComment' instead of 'userName', 'userEmail', and 'userComment'.